sdk/src/main/scala/com/horizen/account/mempool/MempoolMap.scala

```diff
@@ -300,53 +275,56 @@ class MempoolMap(stateReaderProvider: AccountStateReaderProvider) extends Scorex
     }
 
     val execTxsOpt = executableTxs.remove(account)
-    if (execTxsOpt.nonEmpty){
-      execTxsOpt.get.dropWhile {
-        case (nonce, id) => {
-          if (nonce.compareTo(newExpectedNonce) < 0) {
+    if (execTxsOpt.nonEmpty) {
+      execTxsOpt.get
+        .withFilter {
```

Comment:
can you explain more, why we use `.withFilter` instead of `.dropWhile` now?
why is it faster?

Reply:
I used `dropWhile` in the beginning because I wanted to stop checking the nonce after having found the first element with nonce >= newExpectedNonce. In fact, being the collections of txs ordered by nonce, all the subsequent txs must have nonce > newExpectedNonce. The problem is that the `dropWhile` builds a new collection and this is really very expensive. `withFilter` instead doesn't create a new collection, but it is like adding a condition that will be evaluated in the next `foreach`.